addons/base_import/models/base_import.py

```diff
@@ -929,9 +929,10 @@ def do(self, fields, columns, options, dryrun=False):
             for index, column_name in enumerate(columns):
                 if column_name:
                     # Update to latest selected field
-                    exist_records = BaseImportMapping.search([('res_model', '=', self.res_model), ('column_name', '=', column_name)])
+                    exist_records = BaseImportMapping.search([('res_model', '=', self.res_model), ('column_name', '=', column_name)], limit=1)
```

Comment:
The `limit=1` does not seem strictly necessary as there should only be one record per construction, but I guess it does not hurt.

Also, some suggestions for improving readability, considering the indent level, the original variable names, and the fact that you were making lines even longer ;-)):
```py
mapping_domain = [('res_model', '=', self.res_model), ('column_name', '=', column_name)]
column_mapping = BaseImportMapping.search(mapping_domain, limit=1)
if column_mapping:
    if column_mapping.field_name != fields[index]:
        column_mapping.field_name = fields[index]
else:
    ....
```